### ISSUE TITLE
TeleportDistance and 'pet call' broken. Pets do not teleport to you - especially when out of range.

### ISSUE THREAD

**Issue by mibby** [Borlea/EchoPet#28]
```
teleportDistance: 100
```

EchoPets dev 5. Pets do not teleport to you when you move past the teleport distance set in the config. Example, spawn in a pet chicken and fly away. It does not teleport closer to you when out of range.

Pet call also does not call the pet to you. Example - spawn in a pet spider then teleport 1,000 blocks away with /tp <name> 1000 100 1000. Calling the pet doesn't bring it to you - at least it isn't visible when doing so. Using /pet hide has particle effects next to you.

**Comment by Borlea** [comment]
Were you flying? Pets do not teleport back to you atm if you are flying(May be something I change, or a config option).

**Comment by mibby** [comment]
I was flying for the testing, yes. I also tested teleporting a far distance and using pet call.

**Comment by Borlea** [comment]
Information for me later: If you fly out of the teleport distance and the pet is no longer loaded(chunks unloaded) the pet will not teleport to you.

The cause is because when flying the pet doesn't follow you because it can't unless it is a pet that can also fly.

**Comment by mibby** [comment]
This still occurs as of dev 11. Pets do not follow you when teleporting to new chunks.

**Comment by JamieSinn** [comment]
Continued in some form here: https://www.ecocitycraft.com/forum/threads/bug-report-pets-in-sg.159791

**Comment by getByteArrayFromHexString** [comment]
@Jamiesinn looks like that is a separate issue caused from the way I handling world changing and pets. But for all I know that could be cause the sg plugin doesn't call proper events

**Comment by Borlea** [comment]
@JamieSinn that could be due to the way I handle world changing with the invisibility changes or due to that plugin not sending proper events

**Comment by mibby** [comment]
EchoPets dev 31
PaperSpigot dev 958 (MC 1.11)

@Borlea Still an issue as of dev 31. 

The pet will teleport with you if it is near you. i.e. being brought with you from one location to /spawn. However, if you manually walk/fly away from it further than the TeleportDistance set, it does not teleport back to you. So for example if you flew more than 100 blocks away from your pet then tried to go to spawn, it does not teleport to spawn with you. As well as ```/pet call`` does not call the pet to you if you are further than the TeleportDistance.

**Comment by Borlea** [comment]
If you go too far from the pet it despawns it while removing the chunk. And this is the case.

I'll look into tricking the api to think the entity was hidden and allow pet call to teleport it.

**Comment by mibby** [comment]
I don't think it is despawning the pet when it is too far away. It just bugs out and sits idly. I could have sworn old behavior dictated that if the pet is greater than the defined distance, it is called back to your side again. You have to manually remove and respawn it for it to function again.

**Comment by JamieSinn** [comment]
Continuing to persist.

**Comment by Borlea** [comment]
The pet despawns.

If you fly too far away from the pet the pet will just stop attempting to walk to you.
After a certain distance the chunk the pet is in will unload causing the pet to 'despawn'. EchoPet still thinks its alive(Now fixed in a commit i'm about to do)

I could possibly figure our a method to call pet.setHidden(true) in so that /pet call will properly spawn back the pet. But I'll see if other solutions exist.

Also with my latest commit I need to find a clean way to remove the pet if isDead is true, and the instance isn't null.

**Comment by JamieSinn** [comment]
Pets no longer appear, but the player is spammed by a message constantly making any sort of play impossible.

https://imgur.com/a/8WVqB#0skoYmr

**Comment by andrewkm** [comment]
Having same sort of problem here.

Haven't fully looked into it, but can you check this out for us:
https://www.ecocitycraft.com/forum/threads/bug-report-pet-in-sg.160885/

We are currently using echopet build 39 from your Jenkins.

**Comment by mibby** [comment]
> If you fly too far away from the pet the pet will just stop attempting to walk to you.

Ah, okay. It use to just re-teleport to you if too far away. Such as if you went through a door and closed the door on the pet. Then flew away. The pet would just teleport back to you if the TeleportDistance is further than what is configured. 

> I could possibly figure out a method to call pet.setHidden(true) in so that /pet call will properly spawn back the pet. But I'll see if other solutions exist.

That would be nice.

**Comment by andrewkm** [comment]
Can confirm this issue with /pet call is now fixed.
